egov-pg-service/src/main/java/org/egov/pg/repository/GatewayMetadata.java

```diff
@@ -0,0 +1,155 @@
+                .moduleName(MDMS_PAYMENT_GATEWAY).build();
+
+
+        List<ModuleDetail> moduleDetails = new LinkedList<>();
+        moduleDetails.add(gatewayModuledetls);
+
+        MdmsCriteria mdmsCriteria = MdmsCriteria.builder().moduleDetails(moduleDetails).tenantId(tenantId)
+                .build();
+
+        MdmsCriteriaReq mdmsCriteriaReq = MdmsCriteriaReq.builder().mdmsCriteria(mdmsCriteria)
+                .requestInfo(requestInfo).build();
+        return mdmsCriteriaReq;
+    }
+
+    public HashMap mDMSCall(RequestInfo requestInfo, String tenantId) {
+
+        MdmsCriteriaReq mdmsCriteriaReq = getMDMSRequest(requestInfo, tenantId);
+        HashMap result = restTemplate.postForObject(getMDMSURL(), mdmsCriteriaReq, HashMap.class);
+        return result;
+    }
+
+    //returns metData for gateway, tenant,module
+    public Map metaData(RequestInfo requestInfo, String gateway, String tenantId, String module) throws Exception {
+        Map gatewayData = mDMSCall(requestInfo, tenantId);
+
+        List gatewayDetails = (List) ((HashMap) ((HashMap) gatewayData.get(MDMS_RESPONSE))
+                .get(MDMS_PAYMENT_GATEWAY)).get(TENANT_GATEWAY_DETAILS);
+        Map result = new HashMap();
+        try {
+            for (int i = 0; i < gatewayDetails.size(); i++) {
+                //if default gateway is needed
+                if (gateway.equals(GATEWAY_DEFAULT) && ((HashMap) gatewayDetails.get(i)).get(GATEWAY_DEFAULT).equals(true)
+                        && ((HashMap) gatewayDetails.get(i)).get(GATEWAY_ENABLED).equals(true)) {
+                    result.put(gateway, ((HashMap) ((HashMap) gatewayDetails.get(i)).get(GATEWAY_DETAILS)).get("*"));
+                    if (((HashMap) ((HashMap) gatewayDetails.get(i)).get(GATEWAY_DETAILS)).containsKey(module)) {
+                        result.putAll(((HashMap) ((HashMap) ((HashMap) gatewayDetails.get(i)).get(GATEWAY_DETAILS)).get(module)));
+                    }
+
+                } else if (((HashMap) gatewayDetails.get(i)).get(GATEWAY_NAME).equals(gateway) && ((HashMap) gatewayDetails.get(i)).get(GATEWAY_ENABLED).equals(true)) {
+
+                    result.put(gateway, ((HashMap) ((HashMap) gatewayDetails.get(i)).get(GATEWAY_DETAILS)).get("*"));
+                    if (((HashMap) ((HashMap) gatewayDetails.get(i)).get(GATEWAY_DETAILS)).containsKey(module)) {
+                        result.putAll(((HashMap) ((HashMap) ((HashMap) gatewayDetails.get(i)).get(GATEWAY_DETAILS)).get(module)));
+                    }
+
+                }
+
+            }
+
+        } catch (Exception e) {
+            throw new Exception("METADATA_FETCH_ERROR", e);
+        }
+        if (result.size() > 1) {
+            log.error( "Expected to find one gateway for tenant " +
+                    "{}, instead found {}", tenantId, result.size());
+            throw new CustomException("GATEWAY_CONFIG_ERROR","More than one default enabled");
```

Comment:
This check should be there if gateway passed is default or for a given name there are multiple definitions